packages/style-value-parser/src/at-queries-from-tokens/media-query.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+// import {
+//   UNITS_BASED_ON_FONT,
+//   UNITS_BASED_ON_ABSOLUTE_UNITS,
+//   Length,
+// } from '../css-types-from-tokens/length';
+
+import { TokenParser } from '../core2';
+import type {
+  TokenAtKeyword,
+  TokenDelim,
+  TokenDimension,
+} from '@csstools/css-tokenizer';
+
+// const mediaQueryLengthParser = Length.parser.where(
+//   (length): implies length is Length =>
+//     UNITS_BASED_ON_ABSOLUTE_UNITS.includes(length.unit) ||
+//     UNITS_BASED_ON_FONT.includes(length.unit),
+// );
+
+export class MediaQuery {
+  queries: OrSeparatedMediaRules;
+  constructor(queries: this['queries']) {
+    this.queries = queries;
+  }
+  toString(): string {
+    return `@media ${this.queries.toString()}`;
+  }
+  static get parser(): TokenParser<MediaQuery> {
+    return TokenParser.sequence(
+      TokenParser.tokens.AtKeyword.where(
+        (token: TokenAtKeyword): implies token is TokenAtKeyword =>
+          token[4].value === 'media',
+      ),
+      OrSeparatedMediaRules.parser,
+    )
+      .separatedBy(TokenParser.tokens.Whitespace)
+      .map(([_at, queries]) => new MediaQuery(queries));
+  }
+}
+
+export class MediaRule {
+  rules: NotMediaRule | MediaQuerySinglePair | MediaQuerySingleWordCondition;
+  constructor(rules: this['rules']) {
+    this.rules = rules;
+  }
+  toString(): string {
+    return this.rules.toString();
+  }
+  static get parser(): TokenParser<MediaRule> {
+    return TokenParser.oneOf(
+      NotMediaRule.parser.map((notRule) => new MediaRule(notRule)),
+      MediaQuerySinglePair.parser.map(
+        (singleRule) => new MediaRule(singleRule),
+      ),
+      MediaQuerySingleWordCondition.parser.map(
+        (singleRule) => new MediaRule(singleRule),
+      ),
+    );
+  }
+}
+
+export class MediaQueryKeywords {
+  key: 'screen' | 'print';
+  constructor(key: this['key']) {
+    this.key = key;
+  }
+  toString(): string {
+    return `@${this.key}`;
```

Comment:
We shouldn't be prepending with `@` here

```suggestion
    return `${this.key}`;
```